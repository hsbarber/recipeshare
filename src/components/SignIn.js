import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { PasswordForgetLink } from './PasswordForget';
import { SignUpLink } from './SignUp';
import { auth } from '../firebase';
import * as routes from '../constants/routes';

const SignInPage = ({ history, users, changeUser }) =>
  <div className="container-bg">
    <div className="form--container">
      <div className="form">
        <h2>Sign In</h2>
        <SignIn history={history} users={users}
        changeUser={changeUser}
        />

      </div>
    </div>
  </div>
const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

const signInState = {
  email: '',
  password: '',
  error: null,
};

class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = { ...signInState};
  }
  // componentDidUpdate() {
  //   const { users } = this.props;
  //     users.forEach(user => {
  //         if ( this.props.authCopyUser !== null && user.email === this.props.authCopyUser.email) {
  //           //this.setState({ user: user.username });
  //           console.log(user.username)
  //           {() => this.props.changeUser(user.username)}
  //         }
  //     })
  // }

  onSubmit = (event) => {
    event.preventDefault();
    const {
      email,
      password,
    } = this.state;
    const {
      history,
      users,
      changeUser,
    } = this.props;

    auth.doSignInWithEmailAndPassword(email, password)
    .then((authUser) => {
      this.setState({ ...signInState });
      if (authUser) {
        console.log(users);
        users.forEach(user => {
          if (user.email === authUser.email) {
              changeUser(user.username, user.email)
          }
        })
      }

    })
    .then(() => {
     history.push(routes.HOME);
    })
    .catch(error => {
      this.setState(byPropKey('error', error));
    });


  }

  render() {
    const {
      email,
      password,
      error,
    } = this.state;

    const isInvalid =
      password === '' ||
      email === '';

    return (
      <form className="form--inputs" onSubmit={this.onSubmit}>
        <h4>Welcome Back, sign in to continue to RecipeShare</h4>
        <input
          value={email}
          onChange={event => this.setState(byPropKey('email', event.target.value))}
          type="text"
          placeholder="Email Address"
        />
        <input
          value={password}
          onChange={event => this.setState(byPropKey('password', event.target.value))}
          type="password"
          placeholder="Password"
        />
        <button disabled={isInvalid} type="submit">
          Sign In
        </button>
        <PasswordForgetLink />
        <SignUpLink />
        { error && <p>{error.message}</p> }
      </form>
    );
  }
}

export default withRouter(SignInPage);

export {
  SignIn,
};