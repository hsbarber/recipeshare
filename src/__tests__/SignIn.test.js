import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignIn from '../components/SignIn';
import '../enzyme';

const getDefaultProps = () => ({
  changeUser: jest.fn(),
  users: [
    {
      email: 'hotmale@hotmail.com',
      username: 'coolguy420',
    },
    {
      email: 'mark@facebook.com',
      username: 'thezuck',
    },
    {
      email: 'elon@tesla.com',
      username: 'spaceshipjesus',
    },
  ],
});
describe('Sign In', () => {
  const { changeUser, users } = getDefaultProps();
  it('shallow renders without crashing', () => {
    shallow(<SignIn changeUser={changeUser} users={users} />);
  });
  it('component matches snapshot', () => {
    const wrapper = shallow(<SignIn changeUser={changeUser} users={users} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('onchange changes state', () => {
    const wrapper = mount(
      <MemoryRouter>
        <SignIn changeUser={changeUser} users={users} />
      </MemoryRouter>
    );
    const firstInput = wrapper.find('.form--inputs input').first();
    firstInput.instance().value = 'catlover@gmail.com';
    firstInput.simulate('change');
    wrapper.update();
    // expect(firstInput.state().value).toEqual('catlover@gmail.com');
    // expect(wrapper.state().email).toEqual('catlover@gmail.com');
    // firstInput.simulate('change', { target: { value: 'catlover@gmail.com' } });
    console.log(wrapper.state());
    //    expect(wrapper.state().email).to.equal('catlover@gmail.com');
    // wrapper.update(); // update wrapper to reflect the setState as component is updated again after setState
    // expect(wrapper.state().email).toEqual('catlover@gmail.com');
  });
});
