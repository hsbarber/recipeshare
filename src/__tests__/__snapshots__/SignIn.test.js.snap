// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(SignIn)
    changeUser={[MockFunction]}
    users={
      Array [
        Object {
          "email": "hotmale@hotmail.com",
          "username": "coolguy420",
        },
        Object {
          "email": "mark@facebook.com",
          "username": "thezuck",
        },
        Object {
          "email": "elon@tesla.com",
          "username": "spaceshipjesus",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
